src/classes/CRLP_Batch_Base_TEST.cls

```diff
@@ -554,6 +554,62 @@ private class CRLP_Batch_Base_TEST {
             'Recurring Donation Nonskew query should not include inner join');
     }
 
+    /**
+     * @description For Account-Contact-Soft-Credit rollups, validate that the FieldsToCheckForNonZero property
+     * returns back a single instance of a numeric field based on the mocked rollups configured for this
+     * rollup type.
+     */
+    @IsTest
+    private static void shouldIdentifySingleFieldToCheckForNonZero() {
+
+        String rollupsJSON = '[' +
+            CMT_UnitTestData_TEST.createRollupRecord('Date Field', null,
+                CMT_UnitTestData_TEST.RollupRecordType.ContactSoftCreditToAccount, 'Matching_Gift_Info_Updated__c',
+                CRLP_Operation.RollupType.Last, 'CloseDate') + ',' +
+
+            CMT_UnitTestData_TEST.createRollupRecord('Numeric Field', null,
+                CMT_UnitTestData_TEST.RollupRecordType.ContactSoftCreditToAccount, 'Matching_Gift_Amount_Min__c',
+                CRLP_Operation.RollupType.LARGEST, 'Amount__c') +
+
+            ']';
+        CRLP_Rollup_SEL.cachedRollups = (List<Rollup__mdt>) JSON.deserialize(rollupsJSON, List<Rollup__mdt>.class);
+
+        CRLP_Batch_Base_NonSkew rollupJob = new CRLP_Account_SoftCredit_BATCH(
+            CRLP_RollupProcessingOptions.RollupTypeFilter.All
+        );
+        List<String> nonZeroFields = rollupJob.fieldsToCheckForNonZero;
+
+        System.assertEquals(1, nonZeroFields.size(), 'There should be one field identified');
+        System.assertEquals(UTIL_Namespace.StrTokenNSPrefix('Matching_Gift_Amount_Min__c'), nonZeroFields[0]);
+    }
+
+    /**
+     * @description For Account-Soft-Credit rollups, validate that the FieldsToCheckForNonZero property
+     * does not find any numeric field based on the mocked rollups configured for this rollup type.
+     */
+    @IsTest
+    private static void shouldIdentifyNFieldsToCheckForNonZero() {
```

Comment:
Definitely good to have the negative test, but I wonder if we should also test for inclusion of numeric field in Account soft credit.

Reply:
Not sure what you mean by this. Can you clarify?

Reply:
This method tests what happens when you do not have any numeric fields identified in an Account soft credit.  If the intent of your two test is to make sure that the behavior is as expected when you do and when you do not have a numeric field, the two tests you have accomplish that.  One of the tests, though tests the inclusion of numeric field in a Contact Soft Credit, and the test for not finding a numeric field is against an Account Soft Credit.  That's probably just fine because you are testing specific results elsewhere.  I was just noting that in the CRLP_Batch_Base test class you have not tested for the inclusion of a numeric field for an Account Soft Credit.  Since you are testing specific behavior elsewhere, I think it's fine to ignore my comment here.

Reply:
Thanks for the explanation earlier. Just to make sure that the `fieldsToCheckForNonZero` is behaving as it should for both types of rollups, I duplicated the unit tests so that these two run for both types.